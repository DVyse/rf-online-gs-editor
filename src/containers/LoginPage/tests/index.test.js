import { merge } from 'lodash';
import { shallow } from 'enzyme';
import React from 'react';

import { LoginPage, mapDispatchToProps } from '../index';

const projectsImportsProcessingData = {
  isProcessing: false,
  countTotal: 0,
  countCompleted: 0,
  countProcesses: 0,
  percent: 0,
};

function getProps(nextProps = {}) {
  return merge(
    {
      dispatch: jest.fn(action => action),
      changeIdent: jest.fn(),
      changePassword: jest.fn(),
      changeLocation: jest.fn(),
      submit: jest.fn(),
      isLoggedIn: false,
      loginPage: {},
      projectsImportsProcessingData,
    },
    nextProps,
  );
}

describe('<LoginPage />', () => {
  it('should exist', () => {
    const props = getProps();
    const wrapper = shallow(<LoginPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render errorMessage', () => {
    const props = getProps({
      loginPage: {
        isError: true,
        errorMessage: 'Error message',
      },
    });
    const wrapper = shallow(<LoginPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should call redirectToMainPageIfIsLoggedIn on didMount & willReceiveProps', () => {
    const spy = jest.spyOn(
      LoginPage.prototype,
      'redirectToMainPageIfIsLoggedIn',
    );
    const props = getProps();
    const wrapper = shallow(<LoginPage {...props} />);
    expect(spy).toHaveBeenCalledTimes(1);
    wrapper.setProps({ isLoggedIn: false });
    expect(spy).toHaveBeenCalledTimes(2);
    spy.mockClear();
  });
  it('should redirectToMainPageIfIsLoggedIn on willReceiveProps', () => {
    const spy = jest.spyOn(
      LoginPage.prototype,
      'redirectToMainPageIfIsLoggedIn',
    );
    const props = getProps();
    const wrapper = shallow(<LoginPage {...props} />);
    wrapper.setProps({ isLoggedIn: true });
    expect(spy).toHaveBeenCalledTimes(2);
    expect(props.changeLocation.mock.calls.length).toBe(1);
    spy.mockClear();
  });
  it('should redirectToMainPageIfIsLoggedIn on didMount', () => {
    const spy = jest.spyOn(
      LoginPage.prototype,
      'redirectToMainPageIfIsLoggedIn',
    );
    const props = getProps({ isLoggedIn: true });
    shallow(<LoginPage {...props} />);
    expect(spy).toHaveBeenCalledTimes(1);
    expect(props.changeLocation.mock.calls.length).toBe(1);
    spy.mockClear();
  });
  it('directly test mapDispatchToProps', () => {
    const dispatch = jest.fn(action => action);
    const dispatchers = mapDispatchToProps(dispatch);

    expect(
      dispatchers.changeIdent({ target: { value: 'test' } }),
    ).toMatchSnapshot();
    expect(
      dispatchers.changePassword({ target: { value: 'test' } }),
    ).toMatchSnapshot();
    expect(dispatchers.changeLocation()).toMatchSnapshot();
    expect(dispatchers.changeLocation('/some-page')).toMatchSnapshot();
    expect(dispatchers.submit()).toMatchSnapshot();
    expect(dispatch.mock.calls.length).toBe(5);
  });
});
