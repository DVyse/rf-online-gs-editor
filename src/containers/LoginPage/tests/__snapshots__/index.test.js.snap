// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> directly test mapDispatchToProps 1`] = `
Object {
  "key": "ident",
  "type": "app/LoginPage/CHANGE_FIELD_VALUE",
  "value": "test",
}
`;

exports[`<LoginPage /> directly test mapDispatchToProps 2`] = `
Object {
  "key": "password",
  "type": "app/LoginPage/CHANGE_FIELD_VALUE",
  "value": "test",
}
`;

exports[`<LoginPage /> directly test mapDispatchToProps 3`] = `
Object {
  "payload": Object {
    "args": Array [
      "/",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`<LoginPage /> directly test mapDispatchToProps 4`] = `
Object {
  "payload": Object {
    "args": Array [
      "/some-page",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`<LoginPage /> directly test mapDispatchToProps 5`] = `
Object {
  "type": "app/LoginPage/SUBMIT",
}
`;

exports[`<LoginPage /> should exist 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      LoginPage
    </title>
    <meta
      content="Description of LoginPage"
      name="description"
    />
  </HelmetWrapper>
  <Header
    currentProject={null}
    currentUser={null}
    isLoggedIn={false}
    projectsImportsProcessingData={
      Object {
        "countCompleted": 0,
        "countProcesses": 0,
        "countTotal": 0,
        "isProcessing": false,
        "percent": 0,
      }
    }
  />
  <div
    className="container is-fluid p-10"
  >
    <FormWrapper>
      <Form
        as="form"
        size="small"
      >
        <FormField>
          <label>
            <FormattedMessage
              defaultMessage="Login or e-mail address"
              id="app.containers.LoginPage.Ident"
              values={Object {}}
            />
          </label>
          <input
            disabled={false}
            onChange={[MockFunction]}
            type="text"
          />
        </FormField>
        <FormField>
          <label>
            <FormattedMessage
              defaultMessage="Password"
              id="app.containers.LoginPage.Password"
              values={Object {}}
            />
          </label>
          <input
            disabled={false}
            onChange={[MockFunction]}
            type="password"
          />
        </FormField>
        <Button
          as="button"
          onClick={[MockFunction]}
          primary={true}
          role="button"
        >
          <FormattedMessage
            defaultMessage="Login"
            id="app.containers.LoginPage.SubmitLoginForm"
            values={Object {}}
          />
        </Button>
      </Form>
    </FormWrapper>
  </div>
</div>
`;

exports[`<LoginPage /> should render errorMessage 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      LoginPage
    </title>
    <meta
      content="Description of LoginPage"
      name="description"
    />
  </HelmetWrapper>
  <Header
    currentProject={null}
    currentUser={null}
    isLoggedIn={false}
    projectsImportsProcessingData={
      Object {
        "countCompleted": 0,
        "countProcesses": 0,
        "countTotal": 0,
        "isProcessing": false,
        "percent": 0,
      }
    }
  />
  <div
    className="container is-fluid p-10"
  >
    <FormWrapper>
      <Notification
        className=""
        type="danger"
      >
        Error message
      </Notification>
      <Form
        as="form"
        size="small"
      >
        <FormField>
          <label>
            <FormattedMessage
              defaultMessage="Login or e-mail address"
              id="app.containers.LoginPage.Ident"
              values={Object {}}
            />
          </label>
          <input
            disabled={false}
            onChange={[MockFunction]}
            type="text"
          />
        </FormField>
        <FormField>
          <label>
            <FormattedMessage
              defaultMessage="Password"
              id="app.containers.LoginPage.Password"
              values={Object {}}
            />
          </label>
          <input
            disabled={false}
            onChange={[MockFunction]}
            type="password"
          />
        </FormField>
        <Button
          as="button"
          onClick={[MockFunction]}
          primary={true}
          role="button"
        >
          <FormattedMessage
            defaultMessage="Login"
            id="app.containers.LoginPage.SubmitLoginForm"
            values={Object {}}
          />
        </Button>
      </Form>
    </FormWrapper>
  </div>
</div>
`;
